bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gnirs/Gnirs.java

```diff
@@ -110,46 +150,38 @@ public Gnirs(GnirsParameters gp, ObservationDetails odp) {
 
-        // GNIRS is spectroscopy only
-        if (_mode instanceof Imaging) {
-            throw new RuntimeException("GNIRS does not support imaging.");
-        }
-
-
         _detector = new Detector(getDirectory() + "/", getPrefix(), "aladdin", "1K x 1K ALADDIN III InSb CCD");
         _detector.setDetectorPixels(DETECTOR_PIXELS);
 
-        _gratingOptics = new GnirsGratingOptics(
-                getDirectory() + "/" + getPrefix(), _grating,
-                _centralWavelength,
-                _detector.getDetectorPixels(),
-                1,
-                isLongCamera() ? LONG_CAMERA_SCALE_FACTOR : 1,
-                1);
-
-        if (_grating.equals(Disperser.D_10) && !isLongCamera())
-            throw new RuntimeException("The grating " + _grating + " cannot be used with the " +
-                    "0.15\" arcsec/pix (Short) camera.\n" +
-                    "  Please either change the camera or the grating.");
-
-        if (!(_filterUsed.equals("none")))
-            if ((_Filter.getStart() >= _gratingOptics.getEnd()) ||
-                    (_Filter.getEnd() <= _gratingOptics.getStart())) {
-                throw new RuntimeException("The " + _filterUsed + " filter" +
-                        " and the " + _grating +
-                        " do not overlap with the requested wavelength.\n" +
-                        " Please select a different filter, grating or wavelength.");
-            }
-
-
+        if (_mode instanceof Spectroscopy) {
+            _gratingOptics = new GnirsGratingOptics(
+                    getDirectory() + "/" + getPrefix(), _grating.get(),
+                    _centralWavelength,
+                    _detector.getDetectorPixels(),
+                    1,
+                    isLongCamera() ? LONG_CAMERA_SCALE_FACTOR : 1,
+                    1);
+
+            if (_grating.equals(Disperser.D_10) && !isLongCamera())
+                throw new RuntimeException("The grating " + _grating.get() + " cannot be used with the " +
+                        "0.15\" arcsec/pix (Short) camera.\n" +
+                        "  Please either change the camera or the grating.");
+
+            if (!_filterUsed.equals("none"))
+                if ((_Filter.getStart() >= _gratingOptics.getEnd()) ||
+                        (_Filter.getEnd() <= _gratingOptics.getStart())) {
+                    throw new RuntimeException("The " + _filterUsed + " filter" +
+                            " and the " + _grating.get() +
+                            " do not overlap with the requested wavelength.\n" +
+                            " Please select a different filter, grating or wavelength.");
+                }
+        }
         addComponent(_detector);
-
-
     }
 
     public edu.gemini.itc.base.Disperser disperser(final int order) {
         return new GnirsGratingOptics(
-                getDirectory() + "/" + getPrefix(), _grating,
+                getDirectory() + "/" + getPrefix(), _grating.get(),
```

Comment:
`disperser` is only called from `GnirsRecipe.calculateSpectroscopy`, which is only called when we the user is asking for spectroscopy (i.e. explicitly defines a disperser), and in that case we're guaranteed to have a value defined for `_grating`.

If you still think the changes need to be made, as the use is restricted to `GnirsRecipe` I could make the method to return `Option<Disperser>` - not too many changes to the code.